### DISCUSSION TITLE
Moving synced scratch org information between dev environments.

### DISCUSSION THREAD

**Discussion by Cerelius** [forcedotcom/cli#2391] [Ideas]
I'm frustrated when trying to work on a single scratch org across multiple environments. I can't move the .sf and .sfdx directories that are generated in the local environment to a different location because they use absolute paths and are encrypted. I'd like to be able to relocate the configuration without losing my source tracking information for the scratch org that I am authenticated to. 

In the past I could do this by moving the .sfdx directory and updating the path to match the new destination. Because the files are encrypted that won't work anymore. 

Ideally they could use a relative pathing instead of an absolute path so the movement to a new directory does not break the configuration. I don't know why the encryption is in place but removing the encryption would be fine from my perspective. Although, I understand there may be reasons for it that I'm not aware of.

[a comment by an automated account is not shown]

**Comment by mshanemc** [comment]
Storing unencrypted Access/Refresh tokens isn't safe enough for Salesforce.

You didn't mention what OS you're on--for example, macos will use its native keychain by default.  

There is an environment variable `SF_USE_GENERIC_UNIX_KEYCHAIN` https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_dev_cli_env_variables.htm that lets you opts out of that behavior so that the CLI creates a `key.json` in the CLI's folder.  That then should be more portable.

We have a different Discussion around making the CLI's home dir more configurable https://github.com/forcedotcom/cli/discussions/2373 which might also help with your portability use case

**Reply by anthony-bath** [reply]
@mshanemc the `SF_USE_GENERIC_UNIX_KEYCHAIN` env var works great for accessing the same scratch org between GitHub Action runs, but even with that and persisting and restoring the locally generated `.sf` and `.sfdx` folders that contain the `localSourceTracking` files, deploys to the org using `sf project deploy start` don't seem to be able to fully utilize that existing source tracking information. What I'm observing in one of our projects  is `sf project deploy start` will be called, and it will sit there for ~15 minutes before an actual deployment starts, but when it finally does it only takes ~20s for the single file I changed in the test.

The whole workflow looks like this:

1. Open a PR
2. GitHub Action runs, clones repo
3. Scratch Org is created
4. Source from repo is deployed
5. `~/.sfdx`, `~/.sf`, `.sf`, and `.sfdx` are persisted
6. Another commit is pushed to the branch
7. GitHub Action runs, clones repo
8. `~/.sfdx`, `~/.sf`, `.sf`, and `.sfdx` are restored
9. Source is deployed - hoping to utilize existing tracking generated in step 4 to deploy only what has changed in the last commit, but seem unable to do so even with `SF_USE_GENERIC_UNIX_KEYCHAIN`. The deploy step of the GitHub Action calling `sf project deploy start` takes approximately as long as the initial deploy, but `sf project deploy start` in this instance only reports ~20-30 seconds.
